Sources/DiskCache/DiskCache.swift

```diff
@@ -7,66 +7,125 @@
 
 import Foundation
 
-/// Provides interfaces for caching and retrieving data to/from disk. This implementation executes
-/// all tasks on the current queue. Consumers should manage dispatching tasks to background queues if needed
+typealias VoidCheckedContinuation = CheckedContinuation<Void, Error>
+
+/// Provides interfaces for caching and retrieving data to/from disk.
 public class DiskCache: Cache {
+    lazy var queue = DispatchQueue.global()
     let storageType: StorageType
-    let appGroupID: String?
 
-    public required init(storageType: StorageType, appGroupID: String? = nil) throws {
+    public required init(storageType: StorageType) throws {
         self.storageType = storageType
-        self.appGroupID = appGroupID
         try createDirectory(directoryURL)
     }
 
-    public func cache(_ data: Data, key: String) throws {
-        try data.write(to: fileURL(key))
+    public func cache(_ data: Data, key: String) async throws {
+        try await withCheckedThrowingContinuation { [weak self] (continuation: VoidCheckedContinuation) -> Void in
```

Comment:
Do you have any plans to switch to `withUnsafeThrowingContinuation` in a future version?

Edit: the docs make it sound like `withCheckedThrowingContinuation` is more for development, but I'm seeing `withCheckedThrowingContinuation` in Apple's sample app, so maybe I misinterpreted the docs

Reply:
No, I didn't plan on it. Is there are reason why we would?

Reply:
My interpretation comes from this part of the docs for [CheckedContinuation](https://developer.apple.com/documentation/swift/checkedcontinuation):

> `CheckedContinuation` performs runtime checks for missing or multiple resume operations. `UnsafeContinuation` avoids enforcing these invariants at runtime because it aims to be a low-overhead mechanism for interfacing Swift tasks with event loops, delegate methods, callbacks, and other non-`async` scheduling mechanisms. However, during development, the ability to verify that the invariants are being upheld in testing is important. Because both types have the same interface, you can replace one with the other in most circumstances, without making other changes. 

I read that as saying there is a performance hit for using checked continuations, it is provided as a convenience during development, but the expectation is that you generally switch to unsafe continuations once you verify your design.